pkg/controllers/management/clusterprovisioner/provisioner.go

```diff
@@ -214,6 +231,32 @@ func (p *Provisioner) Updated(cluster *v3.Cluster) (runtime.Object, error) {
 	return obj.(*v3.Cluster), err
 }
 
+func (p *Provisioner) migrateHelmReleases(cluster *v3.Cluster) (*v3.Cluster, error) {
+	if !features.HelmCleanupOnUpgrade.Enabled() ||
+		cluster.Status.Version == nil ||
+		semver.Compare(cluster.Status.Version.GitVersion, "v1.25") < 0 ||
+		apimgmtv3.ClusterConditionHelmReleasesMigrated.IsTrue(cluster) {
+		// Skip cleaning up releases if any of the conditions are true
+		return cluster, nil
```

Comment:
Nit: 

I would recommend having this check (which determines when the migration function should be ran) as part of the `Updated` function rather than the `migrateHelmReleases` function. This way, the separation of concerns is maintained - this function only knows how to migrate helm releases, where the `Updated` function evaluates when to migrate helm releases.